### DISCUSSION TITLE
how to clear warning/open issue from masthead?

### DISCUSSION THREAD

**Discussion by jmazzitelli** [kiali/kiali#4750] [Q&A]
What does "1 Open Issue" mean and how can I get rid of that warning? After I see the warning, I am OK with clearing it out, but I can't figure out how to make that go away out of the masthead.

![image](https://user-images.githubusercontent.com/2029470/155207947-61eeff2e-6e47-45af-8fe6-85422bf3bac9.png)

**Comment by lucasponce** [comment]
It seems it needs a tweak in that component.
"Open Issues" don't seem that can be "cleared", but "Notifications can be marked as read or delete them".
![image](https://user-images.githubusercontent.com/1662329/155284151-2263ff62-edcc-495f-b6bc-4c3926ab03ac.png)

Perhaps this warning can be a normal "Warning" notification.

**Comment by israel-hdez** [comment]
That current behavior was the suggested UX on 2018. By design, you cannot clear it because it is an ongoing potential problem.

Read here: https://groups.google.com/g/kiali-dev/c/stbG1wM-iAs/m/arvew396AQAJ
And that behavior was applied here: https://github.com/kiali/kiali-ui/pull/632#issue-355438013

I wouldn't change this behavior without involving the UX team.

**Reply by israel-hdez** [reply]
That said, I think nowadays I would prefer to remove the version check in preference to the compatibility matrix in the docs.
We tend to forget to update that version check :-/

**Reply by jshaughn** [reply]
the version check is brand new :)

**Reply by israel-hdez** [reply]
Is it new again? I remember that it is old :-)

**Reply by jmazzitelli** [reply]
Did we have that feature before??? We just had a community member submit/merge the PR on this 2 weeks ago. See:

* https://github.com/kiali/kiali/pull/4690
* https://github.com/kiali/kiali/issues/4604

**Reply by israel-hdez** [reply]
@jmazzitelli Yes, this is where we do version checks (it is MUCH simpler than the work of our community member):

https://github.com/kiali/kiali/blob/fc79fd4ada9f0bab6e6a1cdb18c6efcda32de1a6/status/versions.go#L269-L403

But we always failed to update supported version numbers. The trend was to update it as users reported that the warning was not right. At some point, it got stalled:

https://github.com/kiali/kiali/blob/fc79fd4ada9f0bab6e6a1cdb18c6efcda32de1a6/config/config.go#L34-L39

~~Indeed, I think what you are ~~seeing~~ showing in your screenshots is the OLD version check. The community guy didn't even do any UI changes.~~

This is why I was saying that we forget to update this and that I'd lean towards removing version checks in preference for docs... And the fact that you don't remember that we already had this is... well... you see my point? :-)

I don't think the situation will improve with that new work. Most likely, we will fail to correctly update version checks as Kiali/Istio are released... But we could give it a 2nd try if you want.

**Reply by israel-hdez** [reply]
Hmmmm... OK, after checking the community PR I can see that in your screenshot you are seeing the message from the *new* version check. Sorry for the confusion.

Anyway, we now have two independent version checks :-)

**Reply by jmazzitelli** [reply]
I write up a github issue - I had no idea we already did something like this.

https://github.com/kiali/kiali/issues/4759

**Reply by nrfox** [reply]
> I don't think the situation will improve with that new work. Most likely, we will fail to correctly update version checks as Kiali/Istio are released... But we could give it a 2nd try if you want.

IMO the fact that this was implemented twice means that it's probably a useful feature for the community. I've seen more than a handful of issues on slack/gh that boil down to a version mismatch. If we are going to keep it then we should find a way to be more disciplined about at least reviewing the compat matrix before new releases.
